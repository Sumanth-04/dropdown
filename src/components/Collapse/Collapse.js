import React, { Component } from 'react'
import PropTypes from 'prop-types';

class Collapse extends Component {
	static propTypes = {
		className: PropTypes.string,
	};

	setActiveKey = activeKey => {
		if (!('activeKey' in this.props)) {
		  this.setState({ activeKey });
		}
		this.props.onChange(this.props.accordion ? activeKey[0] : activeKey);
	  };

	onClickItem = key => {
		let activeKey = this.state.activeKey;
		if (this.props.accordion) {
		  activeKey = activeKey[0] === key ? [] : [key];
		} else {
		  activeKey = [...activeKey];
		  const index = activeKey.indexOf(key);
		  const isActive = index > -1;
		  if (isActive) {
			// remove active state
			activeKey.splice(index, 1);
		  } else {
			activeKey.push(key);
		  }
		}
		this.setActiveKey(activeKey);
	  };
	
	render() {
		return (
			<div>
				{this.props.children}
			</div>
		)
	}
}

export default Collapse;