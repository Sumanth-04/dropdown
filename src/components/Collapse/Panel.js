import React, { Component } from 'react'
import PropTypes from 'prop-types';

class Panel extends Component {
	static propTypes = {
		className: PropTypes.string,
	};

	constructor(){
		super()
		this.state = {
			showPanelBody : false
		}
	}

	togglePanelBody(){
		this.setState({showPanelBody : !this.state.showPanelBody})
	}

	render() {
		return (
			<div>
				<div onClick={() => this.togglePanelBody()}>
					<span>{this.props.header}</span>
				</div>
				<div style={{display : this.state.showPanelBody ? 'block' : 'none'}}>
					{this.props.children}
				</div>
			</div>
		)
	}
}

export default Panel;