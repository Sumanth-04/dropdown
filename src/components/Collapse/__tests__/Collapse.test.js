import React from "react";
import renderer from 'react-test-renderer';
import Collapse from "../Collapse";
import Panel from "../Panel";
//import {isTSAnyKeyword} from '@babel/types';

it("Collapse should have children and Content in each child", () => {
  const tree = renderer
		.create(
			<Collapse className="collapse" accordion={true}>
      <Panel header="Collapse Here >" headerClass="my-header-class">
        this is panel content
      </Panel>
      <Panel header="Collapse Here >">this is panel content2</Panel>
    </Collapse>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
  
});
