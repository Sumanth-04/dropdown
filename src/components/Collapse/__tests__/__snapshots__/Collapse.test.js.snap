// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse should have children and Content in each child 1`] = `
<div>
  <div>
    <div
      onClick={[Function]}
    >
      <span>
        Collapse Here &gt;
      </span>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      this is panel content
    </div>
  </div>
  <div>
    <div
      onClick={[Function]}
    >
      <span>
        Collapse Here &gt;
      </span>
    </div>
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      this is panel content2
    </div>
  </div>
</div>
`;
